Source/Charts/Utils/ViewPortHandler.swift

```diff
@@ -250,160 +227,91 @@ open class ViewPortHandler: NSObject
+        touchMatrix = newMatrix
         
         // make sure scale and translation are within their bounds
-        limitTransAndScale(matrix: &_touchMatrix, content: _contentRect)
+        limitTransAndScale(matrix: &touchMatrix, content: contentRect)
         
         chart.setNeedsDisplay()
         
-        return _touchMatrix
+        return touchMatrix
     }
     
     /// limits the maximum scale and X translation of the given matrix
-    private func limitTransAndScale(matrix: inout CGAffineTransform, content: CGRect?)
+    private func limitTransAndScale(matrix: inout CGAffineTransform, content: CGRect)
     {
         // min scale-x is 1
-        _scaleX = min(max(_minScaleX, matrix.a), _maxScaleX)
+        scaleX = min(max(minScaleX, matrix.a), maxScaleX)
         
         // min scale-y is 1
-        _scaleY = min(max(_minScaleY,  matrix.d), _maxScaleY)
+        scaleY = min(max(minScaleY,  matrix.d), maxScaleY)
         
+        let width = content.width
+        let height = content.height
+
+        let maxTransX = -width * (scaleX - 1.0)
+        transX = min(max(matrix.tx, maxTransX - transOffsetX), transOffsetX)
         
-        var width: CGFloat = 0.0
-        var height: CGFloat = 0.0
-        
-        if content != nil
-        {
-            width = content!.width
-            height = content!.height
-        }
-        
-        let maxTransX = -width * (_scaleX - 1.0)
-        _transX = min(max(matrix.tx, maxTransX - _transOffsetX), _transOffsetX)
-        
-        let maxTransY = height * (_scaleY - 1.0)
-        _transY = max(min(matrix.ty, maxTransY + _transOffsetY), -_transOffsetY)
+        let maxTransY = height * (scaleY - 1.0)
+        transY = max(min(matrix.ty, maxTransY + transOffsetY), -transOffsetY)
         
-        matrix.tx = _transX
-        matrix.a = _scaleX
-        matrix.ty = _transY
-        matrix.d = _scaleY
+        matrix.tx = transX
+        matrix.a = scaleX
+        matrix.ty = transY
+        matrix.d = scaleY
     }
     
     /// Sets the minimum scale factor for the x-axis
     @objc open func setMinimumScaleX(_ xScale: CGFloat)
     {
-        var newValue = xScale
-        
-        if newValue < 1.0
-        {
-            newValue = 1.0
-        }
-        
-        _minScaleX = newValue
-        
-        limitTransAndScale(matrix: &_touchMatrix, content: _contentRect)
+        minScaleX = max(xScale, 1)
+        limitTransAndScale(matrix: &touchMatrix, content: contentRect)
     }
     
     /// Sets the maximum scale factor for the x-axis
     @objc open func setMaximumScaleX(_ xScale: CGFloat)
     {
-        var newValue = xScale
-        
-        if newValue == 0.0
-        {
-            newValue = CGFloat.greatestFiniteMagnitude
-        }
-        
-        _maxScaleX = newValue
-        
-        limitTransAndScale(matrix: &_touchMatrix, content: _contentRect)
+        maxScaleX = xScale == 0 ? .greatestFiniteMagnitude : xScale
```

Comment:
is it a  good practice to use 0 as hint to  CGFloat?

Reply:
`0 == 0.0`. I think the reason `0.0` is used elsewhere is carryover from Objective-C

Reply:
yes, 0.0 looks like more natural as float, while 0 gives me an int imply..